lib/kafka/offset_manager.rb

```diff
@@ -111,17 +116,51 @@ def seconds_since_last_commit
       Time.now - @last_commit
     end
 
+    def seconds_since_last_recommit
+      Time.now - @last_recommit
+    end
+
     def committed_offset_for(topic, partition)
       @committed_offsets ||= @group.fetch_offsets
       @committed_offsets.offset_for(topic, partition)
     end
 
+    def offsets_to_commit(recommit = false)
+      if recommit && !@committed_offsets.nil?
```

Comment:
I think I can add a spec that at least makes it clearer that calling `#next_offset_for` repopulates `@committed_offsets`.